modmail/extensions/utils/error_handler.py

```diff
@@ -162,30 +167,36 @@ async def on_command_error(self, ctx: commands.Context, error: commands.CommandE
                 msg = f"Command `{ctx.invoked_with}` is disabled."
                 if reason := ctx.command.extras.get("disabled_reason", None):
                     msg += f"\nReason: {reason}"
-                embed = self.error_embed(msg, title="Command Disabled")
+                embed = self.error_embed("Command Disabled", msg or str(error))
 
         elif isinstance(error, commands.CommandInvokeError):
-            # generic error
-            logger.error(f'Error occurred in command "{ctx.command}".', exc_info=error.original)
-            # todo: this should log somewhere else since this is a bot bug.
-            embed = self.error_embed(
-                "Oops! Something went wrong internally in the command you were trying to execute. "
-                "Please report this error and what you were trying to do to the developers."
-            )
             if isinstance(error.original, discord.Forbidden):
+                logger.warn(f"Permissions error occurred in {ctx.command}.")
                 await self.handle_bot_missing_perms(ctx, error.original)
-                return
+                should_respond = False
+            else:
+                # generic error
+                logger.error(f'Error occurred in command "{ctx.command}".', exc_info=error.original)
+                # todo: this should log somewhere else since this is a bot bug.
+                embed = self.error_embed(
+                    "Error Occurred",
+                    "Oops! Something went wrong internally in the command you were trying to execute. "
+                    "Please report this error and what you were trying to do to the developers.",
+                )
```

Comment:
Is this always going to be a bot bug, or will plugins fire this too?

Reply:
Plugins and internal features will fire this, so I've customised it in 34e7696 to detect the name of the module it was fired from.